httpclient5/src/main/java/org/apache/hc/client5/http/utils/Base64.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.client5.http.utils;
+
+
+import java.nio.charset.StandardCharsets;
+
+import static java.util.Base64.getEncoder;
+import static java.util.Base64.getMimeDecoder;
+
+/**
+ * Provide implementations of the Base64 conversion methods from commons-codec, delegating to  the Java Base64
+ * implementation.
+ * <p>
+ * Notes:
+ * <p>
+ * <ul>commons-codec accepts null inputs, so this is also accepted here. This is not done in the Java 8 implementation</ul>
+ * <ul>decoding invalid inputs returns an empty value. The Java 8 implementation throws an exception, which is caught here</ul>
+ * <ul>commons-codec decoders accept both standard and url-safe variants of input. This needs to be remapped for Java 8, as it
+ * will accept either one or the other, but not both. This is likely a rarely-needed requirement, but is provided to avoid
+ * compatibility surprises.
+ * </ul>
+ * <ul>Only the features currently used by http-client are implemented here rather than all the features of commons-coded</ul>
+ */
+
+public class Base64 {
```

Comment:
@j3graham Please mark `Base64` and `Hex` as `@Internal`. They should not be a part of supported public APIs.